assets/js/wp-consent-api.js

```diff
@@ -18,33 +18,56 @@ window.waitfor_consent_hook = consent_api.waitfor_consent_hook;
 /**
  * Check if a user has given consent for a specific category.
  *
- * @param {string} category The category to check consent against.
+ * @param {string} item The item to check consent against.
+ * @param {string} type category or service
  */
-function wp_has_consent(category) {
-    var consent_type;
-    if (typeof (window.wp_consent_type) !== "undefined"){
+function wp_has_consent(item, type= 'category') {
+	let has_consent = false;
+	console.log("has consent check for "+item);
+	//for service consent, we start checking if the service's category already has consent. If so, return true and bail.
+	if ( 'service' === type ) {
+		let category = wp_get_service_category( item );
+		if ( wp_has_consent(category) ) {
+			has_consent = true;
+		}
+	}
+
+	let consent_type;
+    if ( typeof (window.wp_consent_type) !== "undefined" ){
         consent_type = window.wp_consent_type;
     }  else {
         consent_type = window.wp_fallback_consent_type
     }
 
-    var has_consent_level = false;
-    var cookie_value = consent_api_get_cookie(consent_api.cookie_prefix + '_' + category);
-
-    if (!consent_type) {
+    let cookie_value = consent_api_get_cookie(consent_api.cookie_prefix + '_' + item);
+    if ( !consent_type ) {
         //if consent_type is not set, there's no consent management, we should return true to activate all cookies
-        has_consent_level = true;
-
+        has_consent = true;
     } else if (consent_type.indexOf('optout') !== -1 && cookie_value === '') {
         //if it's opt out and no cookie is set we should also return true
-        has_consent_level = true;
-
+		has_consent = true;
     } else {
         //all other situations, return only true if value is allow
-        has_consent_level = (cookie_value === 'allow');
+		has_consent = (cookie_value === 'allow');
     }
 
-    return has_consent_level;
+    return has_consent;
+}
+
+/**
+ * Retrieve the category of a registered service
+ *
+ * @param service
+ * @returns {string}
+ */
+function wp_get_service_category( service ) {
+	let services = consent_api.services;
+	services.forEach(function(service) {
+		if ( service.name === service ) {
```

Comment:
The parameter of the function and the closure collides: both are called "service"

Reply:
good one, fixed.

Reply:
TypeScript's `tsc` has static analysis built in.
https://github.com/szepeviktor/byte-level-care/issues/245